sdks/java/core/src/main/java/org/apache/beam/sdk/io/AvroSchemaIOProvider.java

```diff
@@ -74,17 +80,25 @@ public boolean requiresDataSchema() {
 
   @Override
   public PCollection.IsBounded isBounded() {
+    // This supports streaming now as well but there's no option for this. The move to
+    // SchemaTransform will remove the need to provide this.
     return PCollection.IsBounded.BOUNDED;
   }
 
   /** An abstraction to create schema aware IOs. */
   private static class AvroSchemaIO implements SchemaIO, Serializable {
     protected final Schema dataSchema;
     protected final String location;
+    protected final Duration windowSize;
```

Comment:
nit: the null checker should complain about this, but it looks like it's disabled in this file (`SuppressWarnings("nullness")` above)
```suggestion
    protected final @Nullable Duration windowSize;
```

Doesn't necessarily need to happen now, we could enable the null checker and fix this issue and others at a later date.

Reply:
Done.